src/nethsec/dpi/__init__.py

```diff
@@ -0,0 +1,302 @@
+
+    Args:
+      - e_uci: euci instance
+
+    Returns:
+        list of dicts, each dict contains the property "config-name", "description", "enabled", "interface" and "blocks"
+    """
+    rules = list[dict[str]]()
+    fetch_rules = utils.get_all_by_type(e_uci, 'dpi', 'rule')
+
+    if not fetch_rules:
+        return rules
+
+    for rule_name in fetch_rules.keys():
+        # skipping rules with criteria, must be custom entries
+        if e_uci.get('dpi', rule_name, 'criteria', default=None) is None:
+            # load blocklist of applications and protocols
+            blocklist = __load_blocklist()
+            # get content of rule
+            rule = fetch_rules[rule_name]
+            # prepare the data to append to rules
+            data_rule = dict[str]()
+            data_rule['config-name'] = rule_name
+            data_rule['enabled'] = rule.get('enabled', '1') == '1'
+            data_rule['interface'] = rule.get('interface', '*')
+            # get the blocked applications/protocols
+            data_rule['blocks'] = list[dict[str]]()
+
+            # filter by application
+            application_blocklist = [item for item in blocklist if item['type'] == 'application']
+            for application in rule.get('application', []):
+                found_app = [item for item in application_blocklist if
+                             item['name'] == application.removeprefix('netify.')]
+                # there's a possibility of not finding the application due to manual edit of the config
+                if len(found_app) > 0:
+                    data_rule['blocks'].append(found_app[0])
+
+            # filter by protocol
+            protocol_blocklist = [item for item in blocklist if item['type'] == 'protocol']
+            for protocol in rule.get('protocol', []):
+                found_protocol = [item for item in protocol_blocklist if item['name'] == protocol]
+                # there's a possibility of not finding the protocol due to manual edit of the config
+                if len(found_protocol) > 0:
+                    data_rule['blocks'].append(found_protocol[0])
+
+            # append rule
+            rules.append(data_rule)
+
+    return rules
+
+
+def __generate_rule_name(e_uci):
+    counter = 0
+    for role in utils.get_all_by_type(e_uci, 'dpi', 'rule').keys():
+        if f'rule{counter}' == role:
+            counter += 1
+    return f'rule{counter}'
+
+
+def __save_rule_data(e_uci: EUci, config_name: str, enabled: bool, interface: str, applications: list[str],
+                     protocols: list[str]):
+    e_uci.set('dpi', config_name, 'enabled', enabled)
+    e_uci.set('dpi', config_name, 'interface', interface)
+    e_uci.set('dpi', config_name, 'application', [f'netify.{application}' for application in applications])
+    e_uci.set('dpi', config_name, 'protocol', protocols)
+
+
+def add_rule(e_uci: EUci, enabled: bool, interface: str, applications: list[str], protocols: list[str]) -> str:
```

Comment:
Rules can have 4 types of actions: https://github.com/NethServer/nethsecurity/blob/fc08343740793edba25562fd37a3ef962fdfe5f0/packages/ns-dpi/files/dpi-config#L75

The add_rule method must take the action as parameter.